### ISSUE TITLE
Document missing classes

### ISSUE THREAD

**Issue by kocsismate** [php/doc-en#2162]
Currently, the following internal classes are not yet documented:

  * [x] `SensitiveParameterValue` (#2165)
  * [x] `ClosedGeneratorException` (#3065/#2997)
  * [x] `com_safearray_proxy` (#3066) 
  * [x] `DOMNameSpaceNode` (#3060)
  * [x]  `LibXMLError`  (#3063)
  * [x] `PDORow` (#3059)

**Comment by cmb69** [comment]
Yay, [`com_safearray_proxy`](https://github.com/php/php-src/pull/10278)! I'll try to figure out its details.

Maybe @TimWolla wants to tackle `SensitiveParameterValue`?

**Comment by TimWolla** [comment]
> Maybe @TimWolla wants to tackle `SensitiveParameterValue`?

I can write the description, but I need someone more experienced to bootstrap the docs for me. I've tried `php doc-base/scripts/docgen/docgen.php -c SensitiveParameterValue`, but that result is vastly incomplete (missing `final` on the class, missing properties, …).

**Comment by kocsismate** [comment]
@TimWolla as soon aa docgen created the file, you can run gen_stub.php (with --replace-classsynopses) and it will fix signature.

**Comment by TimWolla** [comment]
I've tried `php build/gen_stub.php --replace-classsynopses ../phpdoc/en/language/predefined/`, but that did not do anything. Am I holding it wrong?

The docgen results are:

```
$ g d HEAD^..HEAD --stat
 language/predefined/interfaces.xml                        |  1 +
 language/predefined/sensitiveparametervalue.xml           | 65 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 language/predefined/sensitiveparametervalue/construct.xml | 69 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 language/predefined/sensitiveparametervalue/debuginfo.xml | 55 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 language/predefined/sensitiveparametervalue/getvalue.xml  | 55 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 language/predefined/versions.xml                          |  4 ++++
 6 files changed, 249 insertions(+)
```

**Comment by kocsismate** [comment]
@TimWolla yes, you are missing a parameter: after the --replace.... You need the stub path(s) in php-src. And the very last parameter is for the doc-en path.

**Comment by TimWolla** [comment]
Thanks, that did something.

**Comment by Girgias** [comment]
> > Currently, the following internal classes are not yet documented:
> > 
> > * [ ]    `LibXMLError`
> 
> LibXMLError is already documented as [libXMLError](https://github.com/php/doc-en/blob/master/reference/libxml/libxmlerror.xml). Is that OK or should one of those be changed to match the spelling of the other?

We should fix the name in the docs to match what the stubs is, which is ``LibXMLError``

**Comment by kocsismate** [comment]
> LibXMLError is already documented as [libXMLError](https://github.com/php/doc-en/blob/master/reference/libxml/libxmlerror.xml).

Thanks for the finding! I've just filed https://github.com/php/doc-en/pull/3063 to fix the issue with the spelling.

**Comment by Girgias** [comment]
All missing classes have been documented :fireworks:

**Comment by kocsismate** [comment]
Sorry, I have to reopen the issue as in the meanwhile (since PHP 8.2) a few other newly added classes became undocumented.

**Comment by TimWolla** [comment]
> few other newly added classes became undocumented.

I would write the docs for IntervalBoundary, but that has the problem that it's the first native enum and I believe PHD cannot yet do enums.

**Comment by kocsismate** [comment]
> I would write the docs for IntervalBoundary, but that has the problem that it's the first native enum and I believe PHD cannot yet do enums.

Uhh, that's a problem indeed. Then I'll try to implement it both at gen_stub.php's and phd's side...

**Comment by Girgias** [comment]
> > I would write the docs for IntervalBoundary, but that has the problem that it's the first native enum and I believe PHD cannot yet do enums.
> 
> Uhh, that's a problem indeed. Then I'll try to implement it both at gen_stub.php's and phd's side...

The issue is that we are using an outdated version of DocBooks, and the new enum synopsis tag is only available in the newest prerelease of Docbook 5.2.

When I get back to the UK next week I will try to finalize the migrations for the markup of classes in the current documentations so that one can update the DocBook version.

**Comment by kocsismate** [comment]
> When I get back to the UK next week I will try to finalize the migrations for the markup of classes in the current documentations so that one can update the DocBook version.

Ah, I completely forgot that you have already been working on this topic. Then I'll leave it up to you to fix phd. :) And I can provide a PR for gen_stub.php if needed.

**Comment by TimWolla** [comment]
What's the status with enum rendering?

**Comment by Girgias** [comment]
I'm going to work on this again now, as I've given up on trying to not give huge amount of work to already über outdated translations.
